.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,95 @@
+name: ci-cd
+
+on:
+  push:
+    branches:
+      - master
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - master
+  schedule:
+    - cron: "0 7 * * *" # runs every day at 7 a.m
+
+jobs:
+  build:
+    name: ${{ matrix.IMAGE_NAME }}@${{ matrix.NPM_TAG }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        node_test: [10] # used for the test suite
+        node: [12.18.3]
+        NPM_TAG: ["latest", "next"]
+        IMAGE_NAME:
+          [
+            "theia-full",
+            "theia",
+            "theia-go",
+            "theia-python",
+            "theia-cpp",
+            "theia-rust",
+            "theia-https",
+            "theia-php",
+          ]
+        include:
+          - IMAGE_NAME: "theia-https"
+            ENV_VARS: "-e token="
+            PORT: 10443
+          # Swift and Dart images use a different node version than the rest
+          - IMAGE_NAME: "theia-swift"
+            NPM_TAG: "next"
+            node_test: 10
+            node: 12
+          - IMAGE_NAME: "theia-swift"
+            NPM_TAG: "latest"
+            node_test: 10
+            node: 12
+          - IMAGE_NAME: "theia-dart"
+            NPM_TAG: "next"
+            node_test: 10
+            node: 12
+          - IMAGE_NAME: "theia-dart"
+            NPM_TAG: "latest"
+            node_test: 10
+            node: 12
```

Comment:
nit: `latest` first like the [`matrix`](https://github.com/theia-ide/theia-apps/pull/446/files#diff-662dff05c7dfe6681c1007ae601e8573d413a03f9f9d53d951c2cb99caba4fd4R23).

Reply:
nit: add `newlines` in between so it reads easier?